File: Python/closest-numbers.py
def closestNumbers(arr):
    arr = sorted(arr)
    ans = []
    mini = math.inf
    for i in range(len(arr)-1):
        diff = arr[i+1]-arr[i]
        if diff < mini:
            ans.clear()
            mini = diff
            
        if diff == mini:
            ans.append(arr[i])
            ans.append(arr[i+1])
        
    return ans

File: Java/closest-numbers.java
class Result{
    public static List<Integer> closestNumbers(List<Integer> arr) {
        Collections.sort(arr);
        int min_diff = Integer.MAX_VALUE;
        ArrayList <Integer> ar = new ArrayList<>();
        for (int i = 0; i < arr.size() - 1; i++) {
            int result = arr.get(i + 1) - arr.get(i);

            if (result < min_diff) {
                ar.clear();
                min_diff = result;
            }

            if (result == min_diff) {
                ar.add(arr.get(i));
                ar.add(arr.get(i + 1));
            }
        }
        return ar;
    }
}